### PR TITLE
The Quartermaster Hardsuit

### PR DESCRIPTION
**Pull request by Endecc**
## About the PR

Added the ~Quartermaster~ luxury mining hardsuit, a requested (and debated) hardsuit specially made for a Quartermaster*.
*ᴺᵒᵗ ˢᵖᵉᶜᶦᶠᶦᶜᵃˡˡʸ ᶠᵒʳ ᵗʰᵉ ᵠᵘᵃʳᵗᵉʳᵐᵃˢᵗᵉʳ.
Does **not** add to any roundstart lockers.
Added a new cargo order, the ~Quartermaster's~ luxury mining hardsuit, at 15,000 Spesos.

## Why / Balance

_The following was for when it was labelled as the Quartermaster's specifically. I can't be bothered to reedit it all._
- **Quartermaster is a desk job, why a hardsuit?** - People have now and again asked for a QM hardsuit, but it's hotly debated because their job doesn't really need it. Salvagers have hardsuits a QM could steal, they have EVA access themselves, or they could outright buy an EVA package themselves. These are all valid points against a roundstart hardsuit, and I agree it's not really necessary in such an implementation. That's why I went a _different_ route.
- **Order it, just to say you could** - The Quartermaster does not start with this hardsuit. Instead, it must be ordered for 15,000 spesos, as a flaunt of department wealth. It's not a particularly good hardsuit, it's slightly faster than its underling Mining hardsuit, but offers roughly a THIRD of the protection
- **It's all in character** - Cargo's whole internal shtick is to make a lot of money, and flaunt it. But there's never really been one good goal item specifically for cargo. Centcomm has heard the pleas of Quartermasters around their sectors, and they have finally listened. Of course, they aren't going to pass up the opportunity to siphon some funds back from you. They know you're good for it.

## Media

_Renamed to "luxury mining hardsuit", sprites remain the same_
qmhardsuit
qmhardsuitcargo

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added the luxury mining hardsuit, a lighter and more versatile mining hardsuit styled after the Quartermaster.
- add: Added luxury mining hardsuit cargo order, because this ELITE cargo suit is too expensive to stock on every station.

### CODE REVIEW

**Comment by lzk228** [comment]
QM don't need it
all their work on station, there's no need to go in space, and even if necessary, there is salvages suits and EVA

**Comment by Endecc** [comment]
> QM don't need it all their work on station, there's no need to go in space, and even if necessary, there is salvages suits and EVA

I specifically addressed these concerns, _point for point_, in the Why / Balance section. It is an obscenely expensive, single hardsuit that does not offer any exceptional protections. They do not _need_ it, no.
However, it is funny. That is why I implemented the way I have as cargo order. Again, it is **not a roundstart hardsuit.**

**Comment by Flareguy** [comment]
The sprite for this is so dumb. I love it.

**Comment by EmoGarbage404** [comment]
If you want a big ticket item to flaunt just port the bicycle crate from /tg/

**Comment by BolloTea** [comment]
In my opinion this is a very stupid idea.  Like a quartermaster suit, like the way you buy it.  With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

**Comment by Aisu9** [comment]
> In my opinion this is a very stupid idea. Like a quartermaster suit, like the way you buy it. With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

another reason to make the request console's catalogue access/id dependant

**Comment by Endecc** [comment]
> If you want a big ticket item to flaunt just port the bicycle crate from /tg/

Back when I made a push to replicate the Advanced Mop function from SS13, I was told that making something new is better than just copying the same stuff from SS13. To my knowledge, this is fulfilling that preference.

> In my opinion this is a very stupid idea. Like a quartermaster suit, like the way you buy it. With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

Compared to the other Head of Department hardsuits, it's really nothing special. If the cargo department seriously wants to shelve the tens of thousands it would take to get multiple of them, why not?

**Comment by Flareguy** [comment]
You should probably leave a comment in the QM locker prototype and on/or on the hardsuit prototype telling people to not put it in the locker, or else someone might end up doing it.

**Comment by mirrorcult** [comment]
do not map suffix probably necessary

**Comment by mirrorcult** [comment]
no problem with this conceptually, its good to have stuff that costs a ton and its funny when its mostly just a status symbol

**Comment by Endecc** [comment]
> do not map suffix probably necessary

Sounds good to me. Added comments to the hardsuit, locker, and helmet.

**Comment by potato1234x** [comment]
i love this

**Comment by EmoGarbage404** [comment]
I still think that this being specifically identified as QM's hardsuit is weird. The concept of a luxury hardsuit is a funny concept for a cargo crate but I think specifically identifying it with the QM is strange. It not only makes it the only head hardsuit to not spawn with the head, but also the only one that has multiple instances and can be easily acquired by any crew member. I think players may find it confusing since it seems to violate the norms that the other head hardsuits establish.

**Comment by Endecc** [comment]
> I still think that this being specifically identified as QM's hardsuit is weird. The concept of a luxury hardsuit is a funny concept for a cargo crate but I think specifically identifying it with the QM is strange. It not only makes it the only head hardsuit to not spawn with the head, but also the only one that has multiple instances and can be easily acquired by any crew member. I think players may find it confusing since it seems to violate the norms that the other head hardsuits establish.

While the order of a general luxury hardsuit is fair, I don't personally see it as funny as Centcomm specifically acknowledging the desires of QMs wanting their own hardsuit, and making them order a specially-designed one at an exorbitant rate just for the privilege of saying they have their own. To me, that's more interesting than something like "This hardsuit is really expensive because it's made out of GOLD", or what-have-you.

The differences from other head hardsuits by having it ordered is very intentional, in that way. It IS violating the norms, because people don't want it to be part of the norm at all, it'd be an unnecessary shift-start boon for a desk job. It's a very intentional departure explicitly to get people to stop asking for a QM hardsuit, because Centcomm doesn't want to hear it anymore. They're getting what they wanted, just not in the most conventional way.

**Comment by Endecc** [comment]
Feedback discussion in discord #contributors agreed that it would be better renamed, so I've done that now with some minor tweaks to the descriptions so it makes sense.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 1]
```diff
@@ -390,6 +390,36 @@
   - type: ToggleableClothing
     clothingPrototype: ClothingHeadHelmetHardsuitSecurityRed
 
+#Quartermaster's Hardsuit
+- type: entity
+  parent: ClothingOuterHardsuitBase
+  id: ClothingOuterHardsuitQm #DO NOT MAP - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
+  name: luxury mining hardsuit
+  description: A refurbished mining hardsuit, fashioned after the Quartermaster. Graphene lining provides less protection, but is much easier to move.
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:398]: You need to remove references (ID, comment, description) to this being related to the QM. Ideally the sprite would be renamed too.

File: Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml [hunk 2]
```diff
@@ -300,6 +300,25 @@
         Piercing: 0.9
         Heat: 0.9
 
+#Quartermaster's Hardsuit
+- type: entity
+  parent: ClothingHeadHardsuitWithLightBase
+  id: ClothingHeadHelmetHardsuitQm #DO NOT MAP - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:306]: Same as above

File: Resources/Prototypes/Catalog/Cargo/cargo_cargo.yml [hunk 3]
```diff
@@ -6,4 +6,14 @@
...
-  group: market
\ No newline at end of file
+  group: market
+
+- type: cargoProduct
+  id: CargoQuartermasterHardsuit
+  icon:
+    sprite: Clothing/Head/Hardsuits/qm.rsi
+    state: icon
+  product: CrateCargoQuartermasterHardsuit
```
Comment (LankLTE) [inline Resources/Prototypes/Catalog/Cargo/cargo_cargo.yml:16]: ^

File: Resources/Locale/en-US/prototypes/catalog/cargo/cargo-cargo.ftl [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ent-CargoQuartermasterHardsuit = { ent-CrateCargoQuartermasterHardsuit }
+    .desc = { ent-CrateCargoQuartermasterHardsuit.desc }
```
Comment (LankLTE) [inline Resources/Locale/en-US/prototypes/catalog/cargo/cargo-cargo.ftl:2]: ^

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ent-CrateCargoQuartermasterHardsuit = luxury mining hardsuit crate
+    .desc = Finally, a hardsuit Quartermasters could call their own. Centcomm has heard you, now stop asking.
```
Comment (LankLTE) [inline Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl:2]: Remove the reference to this being for the QM.

File: Resources/Prototypes/Catalog/Fills/Crates/cargo.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+- type: entity
+  id: CrateCargoQuartermasterHardsuit
+  parent: CrateCommandSecure
+  components:
+  - type: StorageFill
+    contents:
+    - id: ClothingOuterHardsuitQm
```
Comment (LankLTE) [inline Resources/Prototypes/Catalog/Fills/Crates/cargo.yml:7]: As above

File: Resources/Prototypes/Catalog/Fills/Lockers/heads.yml [hunk 7]
```diff
@@ -5,6 +5,7 @@
   components:
   - type: StorageFill
     contents:
+    #DO NOT MAP QUARTERMASTER HARDSUIT LOCKER - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
```
Comment (LankLTE) [inline Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:8]: This can be removed too

**Review by LankLTE — CHANGES_REQUESTED** [review]
Hardsuit needs to internally not be the QMs' too- don't just change the name of the suit.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml (hunk 1 above)
Comment (Endecc) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:398]: I was told leaving description references to the Quartermaster would be fine. I can definitely change the internal thongs for sure, but is it seriously imperative to remove every vein and spine of intent I had with the original PR?

**Comment by EtereohZ** [comment]
Should a "Luxury" category be added to the cargo console if more premium/luxury items are, maybe, going to be added in the future?

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl (hunk 5 above)
Comment (Endecc) [reply Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl:2]: Same as hardsuits.yml comment - removing the spine of the PR

**Comment by Endecc** [comment]
> Should a "Luxury" category be added to the cargo console if more premium/luxury items are, maybe, going to be added in the future?

The cargo category the hardsuit is in is extremely bare-bones already, serving only 1 order for an extra cargo palette. If such a category is made, it wouldn't be just for this. That'd make two categories with 1 item.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml (hunk 1 above)
Comment (LankLTE) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:398]: Just looked through the relevant discord discussion. Description is fine, yeah.

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl (hunk 5 above)
Comment (LankLTE) [reply Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl:2]: This is similarly fine.

**Comment by EmoGarbage404** [comment]
Fix the changelog to reflect the PR + be less wordy

**Comment by Endecc** [comment]
> Fix the changelog to reflect the PR + be less wordy

done

**Review by LankLTE — APPROVED** [review]

**Pull request merged**